### PR TITLE
:seedling: Bump Golang to v1.23.4, CAPI to v1.9.0-rc.1 and, k8s.io/utils

### PR DESCRIPTION
**Pull request by peppi-lotta**
**What this PR does / why we need it**: Bump Golang to v1.23.4, CAPI to v1.9.0-rc.1, controller-runtime to v0.19.3  and, k8s.io/utils

### CODE REVIEW

File: apis/go.mod [hunk 1]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/stretchr/testify v1.10.0
 	k8s.io/api v0.31.3
 	k8s.io/apimachinery v0.31.3
-	sigs.k8s.io/controller-runtime v0.19.2
+	sigs.k8s.io/controller-runtime v0.19.3
```
Comment (tuminoid) [inline apis/go.mod:11]: This is bumped by dependabot already

**Review by tuminoid — COMMENTED** [review]
CR is being bumped by dependabot here https://github.com/metal3-io/baremetal-operator/pull/2106 so you need another rebase soon.

File: apis/go.mod (hunk 1 above)
Comment (peppi-lotta) [reply apis/go.mod:11]: Thanks for the head up . I'll rebase once that PR merges.

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Review by Rozzii — COMMENTED** [review]
/approve

**Pull request merged**